l293x/l293x.go

```diff
@@ -0,0 +1,100 @@
+// Package l293x provides a driver to the L293/L293D H-bridge chip
+// typically used to control DC motors.
+//
+// Datasheet: https://www.ti.com/lit/ds/symlink/l293d.pdf
+//
+package l293x // import "tinygo.org/x/drivers/l293x"
+
+import (
+	"machine"
+)
+
+// Device is a motor without speed control.
+type Device struct {
+	d1, d2 machine.Pin
+	power  machine.Pin
+}
+
+// New returns a new Motor driver for GPIO-only operation.
+func New(d1, d2, p machine.Pin) Device {
+	return Device{
+		d1:    d1,
+		d2:    d2,
+		power: p,
+	}
+}
+
+// Configure configures the Device.
+func (d *Device) Configure() {
+	d.d1.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.d2.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.power.Configure(machine.PinConfig{Mode: machine.PinOutput})
+
+	d.Stop()
+}
+
+// Forward turns motor on in forward direction.
+func (d *Device) Forward() {
+	d.d1.High()
+	d.d2.Low()
+	d.power.High()
+}
+
+// Backward turns motor on in backward direction.
+func (d *Device) Backward() {
+	d.d1.Low()
+	d.d2.High()
+	d.power.High()
+}
+
+// Stop turns motor off.
+func (d *Device) Stop() {
+	d.d1.Low()
+	d.d2.Low()
+	d.power.Low()
+}
+
+// PWMDevice is a motor with speed control.
+type PWMDevice struct {
+	d1, d2 machine.Pin
+	speed  machine.PWM
+}
+
+// NewWithSpeed returns a new PWMMotor driver that uses PWM to control speed.
+func NewWithSpeed(d1, d2 machine.Pin, s machine.PWM) PWMDevice {
```

Comment:
here it's a bit more confusing if we do s => speed, maybe speedPin? or pwn?

Reply:
Did some renaming based on your suggestions, and also matching the datasheet more closely.